pkg/notes/notes_data.go

```diff
@@ -513,3 +532,74 @@ func setEmptyComponentGitHubRepositoryFromOtherKilnfile(k1, k2 cargo.Kilnfile) c
 }
 
 func closeAndIgnoreError(c io.Closer) { _ = c.Close() }
+
+type TrainstatFetcher interface {
+	FetchTrainstatNotes(milestone string, version string, tile string) ([]string, error)
+}
+
+type TrainstatClient struct {
+	host       string
+	httpClient *http.Client
+}
+
+func NewTrainstatClient(host string) TrainstatClient {
+	if host == "null" || strings.TrimSpace(host) == "" {
+		host = TrainstatUrl()
+	}
+	client := &http.Client{
+		Timeout: time.Second * 10,
+	}
+	return TrainstatClient{
+		host:       host,
+		httpClient: client,
+	}
+}
+
+func (t *TrainstatClient) FetchTrainstatNotes(milestone string, version string, tile string) (notes []string, err error) {
+	if !t.tileSupported(tile) {
+		return []string{}, nil
+	}
+
+	baseURL := fmt.Sprintf("%s/%s", t.host, "/api/v1/release_notes")
+	req, err := http.NewRequest(http.MethodGet, baseURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	q := req.URL.Query()
+	q.Set("milestone", milestone)
+	q.Set("version", version)
+	q.Set("tile", tile)
+	req.URL.RawQuery = q.Encode()
+
+	res, err := t.httpClient.Do(req)
```

Comment:
httpClient will not return an error on non success status codes.

Write a check on req.StatusCode after the Do error check.